pkgs/development/python-modules/array-api-compat/default.nix

```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pytestCheckHook,
+  setuptools,
+  numpy,
+  jaxlib,
+  jax,
+  torch,
+  dask,
+  sparse,
+  array-api-strict,
+  config,
+  cudaSupport ? config.cudaSupport,
+  cupy,
+  nix-update-script,
+}:
+
+buildPythonPackage rec {
+  pname = "array-api-compat";
+  version = "1.8";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "data-apis";
+    repo = "array-api-compat";
+    rev = "refs/tags/${version}";
+    hash = "sha256-DZs51yWgeMX7lmzR6jily0S3MRD4AVlk7BP8aU99Zp8=";
+  };
+
+  build-system = [ setuptools ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    numpy
+    jaxlib
+    jax
+    torch
+    dask
+    sparse
+    array-api-strict
+  ] ++ lib.optionals cudaSupport [ cupy ];
+
+  pythonImportsCheck = [ "array_api_compat" ];
+
+  # CUDA (used via cupy) is not available in the testing sandbox
+  checkPhase = ''
+    runHook preCheck
+    python -m pytest -k 'not cupy'
+    runHook postCheck
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
```

Comment:
https://github.com/NixOS/nixpkgs/issues/208242

Reply:
It would be great if reviewers consistently applied this and there was a pointer in the contributing docs instead of just dropping an issue link on PRs.

Reply:
nixpkgs is a huge project with thousands of contributors, not everyone can be across everything, but it's fine, we just trying to do our best